### ISSUE TITLE
Can node_modules and other things be removed after build?

### ISSUE THREAD

**Issue by tremby** [mars/create-react-app-buildpack#169]
I'm chasing issues with slug size.

I note that node_modules is by far the largest directory in the slug (950M uncompressed! and most of this is node_modules/.cache), but I would have thought this isn't even needed in a create-react-app application once the build has finished.

Everything we need to serve is surely now in the build directory, no?

Would it be possible to add a step to the build procedure so that after building, node_modules and certain other things (don't need src and public even, I think) get removed?

**Comment by mars** [comment]
If you’re using this buildpack straight, without any other javascript buildpacks or customizations, then I think it’s okay to delete `node_modules/` at the end of the build process.

In `package.json`, at the end of the build script property, you can add ` && rm -rf node_modules/` to delete it after build.

**Comment by tremby** [comment]
I'll give that a go.

I'm curious: in what sort of circumstances would a create-react-app app still need node_modules? What other JS buildpacks and what sort of customizations are you referring to? I ask because create-react-app-buildpack's purpose seems laser-focused, and as far as I can tell someone using it for its intended purpose should have no need for anything except the dist directory once the build is done.

**Comment by tremby** [comment]
Seems to work great.

But after doing this, node_modules can no longer be part of what Heroku caches between builds, since we've deleted it by the time Heroku tries to cache things. I'm not sure yet how much time this is going to add to the builds. But the slug size for this project went down from ~530M to ~25M so it's likely worth it.

It seems very weird to me that the stuff matched in .slugignore is deleted *before* the build takes place. Not sure if that's to do with the buildpack or to do with Heroku itself. But it seems to me that better behaviour would be to use .slugignore as a mask when producing the slug without actually deleting any files, then the caching could continue as usual.

**Comment by mars** [comment]
Your comments makes sense @tremby, but this personal, open-source buildpack has no influence over things like that. [This buildpack primarily glues other buildpacks together](https://github.com/mars/create-react-app-buildpack/blob/master/README.md#user-content-architecture-) with minimal code of its own.

You might find existing issues about this or could open a new issue over at the [Node buildpack repo](https://github.com/heroku/heroku-buildpack-nodejs).
